cmd/signature-v2.go

```diff
@@ -282,7 +282,13 @@ func canonicalizedResourceV2(encodedPath string, encodedQuery string) string {
 			canonicalQueries = append(canonicalQueries, key)
 			continue
 		}
-		canonicalQueries = append(canonicalQueries, key+"="+val)
+		// Resources values should be unescaped
+		unescapedVal, err := url.QueryUnescape(val)
+		if err != nil {
+			errorIf(err, "Unable to unescape query value (query = `%s`, value = `%s`)", key, val)
+			continue
+		}
+		canonicalQueries = append(canonicalQueries, key+"="+unescapedVal)
```

Comment:
@vadmeste are you sure? because http://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html
```
Elements of the CanonicalizedResource that come from the HTTP Request-URI should be signed literally as they appear in the HTTP request, including URL-Encoding meta characters.
```